### PR TITLE
Add Latin locale

### PR DESCRIPTION
**Pull request by tukusejssirs**

### CODE REVIEW

**Comment by tukusejssirs** [comment]
@emagnier, @pejulian, so far, I have localised all (I presume) the keys from `general.ts` and some of the others I have found on the Internet.

Note that I have marked this PR as WIP because:
1. I’d like to localise all the keys from `en.js`;
2. I have left all my comments in `la.ts` that need to be tended (I have also commented out all the untranslated keys);
3. I used the key names based on PR #202, i.e. this PR should be merged _after_ #202 is merged;
4. I still need to fix #196 in order to make the l10n work.
5. I have removed all the newlines between keys and their values (I prefer it that way because it makes it easier to sort the selected keys).

**Comment by tukusejssirs** [comment]
Just a quick status update: I am halfway through the Latin localisation. :smiley: I must confess, currently I skip some of the keys if I fairly quickly don’t find the Latin translation on the Internet. Most keys can be quite easily found on the Internet (either at vatican.va or Wikipedia or some old Latin documents/books like some biographies/vitæ.

Just for the info: this is how I check the progress: first I get the list of all keys (and just them) in `en.ts`, then I do the same in `la.ts`, compare them, count the keys in both and calculate the percentage of the coverage. The code below might be easier to understand. :smiley:

```bash
# Get the list of `la.ts` keys
grep -hPo "^\s+'*\K[^}': ]+(?='*:)" "src/locales/la.ts" | \
  sed '/^advent$\|^season$\|^$\|^feria$\|^sunday$\|^christmastide$\|^day$\|^octave$\|^epiphany$\|^before$\|^after$\|^ordinaryTime$\|^lent$\|^dayAfterAshWed$\|^holyWeek$\|^eastertide$\|^liturgicalColors$\|^GOLD$\|^GREEN$\|^PURPLE$\|^RED$\|^ROSE$\|^WHITE$\|^celebrations$\|^sanctoral$/d' | \
  sort > la_keys

# Get the list of `en.ts` keys
grep -hPo "^\s+'*\K[^}': ]+(?='*:)" "src/locales/en.ts" | \
  sed '/^advent$\|^season$\|^$\|^feria$\|^sunday$\|^christmastide$\|^day$\|^octave$\|^epiphany$\|^before$\|^after$\|^ordinaryTime$\|^lent$\|^dayAfterAshWed$\|^holyWeek$\|^eastertide$\|^liturgicalColors$\|^GOLD$\|^GREEN$\|^PURPLE$\|^RED$\|^ROSE$\|^WHITE$\|^celebrations$\|^sanctoral$/d' | \
  sort > en_keys

# Create some variables
missing=$(diff -y --suppress-common-lines la_keys en_keys | wc -l)
total=$(cat en_keys | wc -l)
done_percent=$((($total - $missing) * 100 / $total))

# Output the info
echo -e "\n$missing missing of $total ($done_percent % done)"

# Remove the auxiliary files
rm en_keys la_keys


# Example output
357 missing of 714 (50 % done)

**Comment by tukusejssirs** [comment]
I have gone through all the keys, but I have skipped the keys of the most of the blesseds and some of the not easily localisable (and verifiable by Google search). Also, I have left there many comments, many of which are TODOs and some of which mark some duplicates.

**Comment by tukusejssirs** [comment]
I have finished the Latin localisation. Note that I am not 100 % sure that every key translation is absolutely correct (but that said, this version is good enough for our initial version of the Latin locale). I am sure that most of them are correct, as I have sourced them from official documents (Holy See decrees, AAS or Notitiæ), some were sourced from some old biographies (_Vitæ_ in Latin) or old liturgical calendars (pre-Vatican II or just after it, when all the Directories/Ordos were written in Latin). There might be some l10n issues, for which I’ll create (probably tomorrow) a separate issue and place into %backlog in order to solve the potential issues some time in the future [probably when I get a copy of the MR2004 and one or more Latin _Ordo(s)_].

I have intentionally skipped the localisation of `Shrove Monday` and `Shrove Tuesday`, as AFAIK there is no such season as Shrovetide in `cal1969`. The closest season to it is the pre-Vatican II season called Pre-Lenten Time ([Septuagesima](https://en.wikipedia.org/wiki/Septuagesima) in Latin; see esp [Catholic usage after 1969](https://en.wikipedia.org/wiki/Septuagesima#Catholic_usage_after_1969) heading). I think we can safely remove them from all locales, as these keys are not used at all.

I have just removed nearly all my comments from `la.ts`. The only comments that left, are notes that some keys are duplicates (I want to remove them while I deal with #175) and a string that I have not yet localised: `christmastide.day`, which in `en.ts` has the value `{{day}} of Christmastide`). When is this key used?

I’d like to ask you, @emagnier and @pejulian, to review and merge this PR as soon as possible (of course, I need to localise the `chirstmastide.day` key before that), because I’d to tend to the locales from the English language and English-speaking countries perspective (i.e. fix the #175 issue; see [this comment](https://github.com/romcal/romcal/issues/175#issuecomment-665673899)). I believe that fixing #175 issue should be done _before_ the new release @emagnier plans doing this month. And even `la.ts` will be modified by these changes.

---

Anyway, I still have not finished the Latin locale in `dayjs`, as I am not a good programmer and I wanted to finish the Latin locale in romcal first. The Latin locale in `dayjs` should be finished, but I need to create some tests and actually test it if its output is correct (I couldn’t somehow make it work). For more info on this matter see iamkun/dayjs#961 issue and iamkun/dayjs#966 PR.

As the `dayjs` Latin locale is not yet finished, I suggest to use `en-GB` locale for Latin for now. Also I’d like to note that even when the `dayjs` Latin locale is implemented, the week number within season and the weekday number in `eastertide.octave` and `christmastide.octave` will need to be converted into Roman numerals (see the [`romanise()`](https://github.com/iamkun/dayjs/blob/26295a657d1ac68aaab6f786e1724fc89d74c92d/src/locale/la.js#L6) in the `dayjs` locale, which I have found [here](https://stackoverflow.com/a/9083076)).

**Comment by tukusejssirs** [comment]
While building romcal to test the Latin locale, I receive the following errors in the browser console:

```js
la is not supported in romcal's date management library, trying to use la instead Locales.ts:191:14
Failed to load locale data for la. romcal will default to "en" locale data Locales.ts:202:14

Uncaught (in promise) ReferenceError: count is not defined
    anonymous Lodash
    b Locales.ts:262
    s tslib.es6.js:100
    s tslib.es6.js:81
    a tslib.es6.js:74
    a tslib.es6.js:70
    b romcal.bundle.es5.js:1
    n Seasons.ts:707
    s tslib.es6.js:100
    s tslib.es6.js:81
    a tslib.es6.js:74
    a tslib.es6.js:70
    n Seasons.ts:696
    B Seasons.ts:696
    s tslib.es6.js:100
    s tslib.es6.js:81
    a tslib.es6.js:74
    a tslib.es6.js:70
    B Seasons.ts:692
    A Calendar.ts:83
    s tslib.es6.js:100
    s tslib.es6.js:81
    o tslib.es6.js:71
    promise callback*u tslib.es6.js:73
    o tslib.es6.js:71
    promise callback*u tslib.es6.js:73
    o tslib.es6.js:71
    promise callback*u tslib.es6.js:73
    a tslib.es6.js:74
    a tslib.es6.js:70
    A Calendar.ts:73
    fetch Calendar.ts:73
    s tslib.es6.js:100
    s tslib.es6.js:81
    a tslib.es6.js:74
    a tslib.es6.js:70
    fetch romcal.bundle.es5.js:1
    calendarFor index.ts:259
    s tslib.es6.js:100
    s tslib.es6.js:81
    o tslib.es6.js:71
    promise callback*u tslib.es6.js:73
    o tslib.es6.js:71
    promise callback*u tslib.es6.js:73
    a tslib.es6.js:74
    a tslib.es6.js:70
    calendarFor romcal.bundle.es5.js:1
    <anonymous> index.html:50
    promise callback* index.html:39
    EventListener.handleEvent* index.html:27
```

I can’t deal with it.

File: src/locales/la.ts [hunk 1]
```diff
@@ -0,0 +1,797 @@
...
+  },
+  holyWeek: {
+    season: 'Hebdomadæ Sanctæ',
+    feria: '{{day}} Hebdomadæ Sanctæ',
+  },
+  eastertide: {
+    season: 'Paschæ',
+    feria: '{{day}}, hebdomada {{week}} temporis paschalis',
+    sunday: 'Dominica {{week}} Paschæ',
+    octave: 'De die {{count}} infra octavam Paschæ',
```
Comment (emagnier) [inline src/locales/la.ts]: Replace `{{count}}` by `{{day}}` and this localization should work properly.

Comment (tukusejssirs) [reply src/locales/la.ts]: Thanks! I have somewhat copied the `{{count}}` from the `christmastide.octave` value.

**Comment by tukusejssirs** [comment]
Where is `christmastide.day` (`{{day}} of Christmastide`) used? That might need to be localised, but I have no idea when it is used (if ever).

---

I’d like to note that `epiphany.before` is in Latin set to `{{day}} post octavam Nativitatis` (English: `{{day}} after the octave of the Nativity [of the Lord]`), because AFAIK in Latin (and in Slovak) it something like `{{day}} before Epiphany` is not used. I suspect that it used to, but not anymore. And I think the same is true in all locales (in the [_Ordo 2018_](https://www.usccb.org/about/divine-worship/liturgical-calendar/upload/2018cal.pdf) for the US, they just simply have _Christmas Weekday_ on 6 Jan, but that call all ferias _weekdays_).

**Comment by tukusejssirs** [comment]
@emagnier, could you double-check the date format in the Latin Missal (ed. typ. tertia)? Are the day of month numbers written as Roman numerals? If so, are they uppercase (`XX`) or lowercase (`xx`)? Are the month names written with uppercase firet letter (`Ianuarius`) or in all lowercase (`ianuarius`)?

Anyway, is there any other issue with this PR? Or you just simply want to remove the season of Epiphany first and then merge this PR?

**Comment by emagnier** [comment]
Day of month are in arabic format everywhere, except in the table index. Month is lowercase.

Let's merge it! Everything looks good, and also: good work for all this translation! 💪 🎉 🚀

**Comment by tukusejssirs** [comment]
> [@tukusejssirs]
> Where is `christmastide.day` (`{{day}} of Christmastide`) used? That might need to be localised, but I have no idea when it is used (if ever).

@emagnier, you haven’t answered me yet the above question. Is `christmastide.day` even used in the romcal codebase? A quick `grep` in the `src/` folder did not match `christmastide.day`. Anyway, that key is the only one not localised in `la.ts`.

**Pull request merged**